twister: Treat overflows as error on integration_platforms during integration mode

PerMac opened the pull request:
Skips due to overflows on integration_platforms during integretion mode should be treated as errors.

PerMac committed:
twister: Remove redundant line.

`self.instances[filtered_instance.name]` and `filtered_instance`
are already referancing the same object.

Signed-off-by: Maciej Perkowski <Maciej.Perkowski@nordicsemi.no>

PerMac committed:
twister: Refactor "skips to errors if integration"

Export the logic to a separate method for reusability

Signed-off-by: Maciej Perkowski <Maciej.Perkowski@nordicsemi.no>

PerMac committed:
twister: Treat overflows as errors when integration mode is on

Skips due to overflows on integration_platforms during integretion
mode should be treated as errors.

Signed-off-by: Maciej Perkowski <Maciej.Perkowski@nordicsemi.no>

nashif commented:
> The PR exposes an existing issue:
> `3/550 warp7_m4 samples/sensor/fxas21002/sample.sensor.fxas21002 ERROR FLASH overflow but is one of the integration platforms (build)`
> What should we do with it?

you need to fix it before we are able to move forward with this PR.

PerMac committed:
sample: sensor: Change integration_platforms for fxas21002 sample

Builds on warp7_m4 were overflowing flash, hence the change to
the another platform from the allow_list, which is not causing
overflows.

Signed-off-by: Maciej Perkowski <Maciej.Perkowski@nordicsemi.no>

PerMac commented:
I chose another platform from platform_allow as an integration platform. @MaureenHelm should there be an issue raised for samples/sensor/fxas21002 overflowing flash on warp7_m4 which was the integration platform?

MaureenHelm commented:
> I chose another platform from platform_allow as an integration platform for overflowing samples/sensor/fxas21002. @MaureenHelm should there be an issue raised for samples/sensor/fxas21002 overflowing flash on warp7_m4 which was the integration platform?

Yes

nashif commented on scripts/pylib/twister/twisterlib/testplan.py:
```diff
@@ -951,3 +943,15 @@ def _create_build_dir_link(self, links_dir_path, instance):
         self.link_dir_counter += 1
+
+
+def skip_to_error_if_integration(options, instance):
```
skip_to_error is a misleading name, sounds like the function skips to error, please reconsider the function name, maybe `switch_skipped_to_error...`

PerMac replied:
changed

golowanow commented on scripts/pylib/twister/twisterlib/testplan.py:
```diff
@@ -952,3 +943,15 @@ def _create_build_dir_link(self, links_dir_path, instance):
         self.link_dir_counter += 1
+
+
+def change_skip_to_error_if_integration(options, instance):
```
looks like this function might be `class TestInstance` method

PerMac replied:
@golowanow I think it is a good suggestion. I even wanted to apply this, but found the PR merged:) I assume that your comment was just a suggestion and hence no "+/-" review. Therefore, I will leave it as it is without creating a new PR (unless you object).
